plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroHiveFileWriter.java

```diff
@@ -144,6 +148,7 @@ public Closeable commit()
     public void rollback()
     {
         try (rollbackAction) {
+            write.abort();
```

Comment:
Good question. I designed API so that it works ALSO with try-with-resources nice

- expect the close() to always be called (but i see how in this context it is redundant)
- require explicit finish/abort before close, so that almost-correctly-looking code (one that neither calls finish/abort) screams loud that it is not correct, ie nothing written

Those two goals do not require that abort must be followed by close. If we agree on this, i can make abort() release any resources immediately, so that close isn't required.